utils/build-script-impl

```diff
@@ -1154,15 +1161,8 @@ if [[ "${EXPORT_COMPILE_COMMANDS}" ]] ; then
     )
 fi
 
-if [[ "${EXTRA_CMAKE_OPTIONS}" ]] ; then
-    for cmake_opt in $(echo "${EXTRA_CMAKE_OPTIONS}" | tr "," "\n")
-    do
-        COMMON_CMAKE_OPTIONS=(
-            "${COMMON_CMAKE_OPTIONS[@]}"
-            "${cmake_opt}"
-        )
-    done
-fi
+# Convert to an array.
+eval EXTRA_CMAKE_OPTIONS=(${EXTRA_CMAKE_OPTIONS})
```

Comment:
Why not:

```
EXTRA_CMAKE_OPTIONS=("${EXTRA_CMAKE_OPTIONS[@]}")
```

This will make it preserve shell metacharacters.

Consider:

```shell
X='a $b c'
eval Y=(${X})
echo ${Y[@]} # a c
```